src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateEqualsAndGetHashCodeFromMembers/GenerateEqualsAndGetHashCodeFromMembersTests.cs

```diff
@@ -2,41 +2,231 @@
+using Microsoft.CodeAnalysis.CodeStyle;
 using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.CSharp.CodeStyle;
 using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
-using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeActions;
-using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editor.UnitTests;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
 using Microsoft.CodeAnalysis.GenerateEqualsAndGetHashCodeFromMembers;
 using Microsoft.CodeAnalysis.PickMembers;
 using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Testing;
+using Microsoft.VisualStudio.Composition;
+using Microsoft.VisualStudio.LanguageServices;
 using Roslyn.Test.Utilities;
 using Xunit;
 
 namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.GenerateEqualsAndGetHashCodeFromMembers
 {
     using static GenerateEqualsAndGetHashCodeFromMembersCodeRefactoringProvider;
+    using static Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions.AbstractCodeActionOrUserDiagnosticTest;
+    using VerifyCS = CSharpCodeRefactoringVerifier<GenerateEqualsAndGetHashCodeFromMembersCodeRefactoringProvider>;
 
-    public class GenerateEqualsAndGetHashCodeFromMembersTests : AbstractCSharpCodeActionTest
+    [UseExportProvider]
+    public class GenerateEqualsAndGetHashCodeFromMembersTests
     {
+        private class Test : VerifyCS.Test
+        {
+            // Our own cached export factory, where we can mixin the TestPickMembersService type.
+            private static readonly IExportProviderFactory s_factory = ExportProviderCache.GetOrCreateExportProviderFactory(
+                TestExportProvider.EntireAssemblyCatalogWithCSharpAndVisualBasic.WithPart(typeof(TestPickMembersService)));
+
+            public bool UseDialog;
+            public ImmutableArray<string> MemberNames;
+            public Action<ImmutableArray<PickMembersOption>> OptionsCallback;
+            public CompilationOptions CompilationOptions;
+
+            public override AdhocWorkspace CreateWorkspace()
+            {
+                if (!UseDialog)
+                    return base.CreateWorkspace();
+
+                // If we're a dialog test, then mixin our mock and initialize its values to the ones the test asked for.
+                var exportProvider = s_factory.CreateExportProvider();
+                var workspace = new AdhocWorkspace(VisualStudioMefHostServices.Create(exportProvider));
+
+                var service = (TestPickMembersService)workspace.Services.GetService<IPickMembersService>();
+                service.MemberNames = MemberNames;
+                service.OptionsCallback = OptionsCallback;
+
+                return workspace;
+            }
+
+            protected override CompilationOptions CreateCompilationOptions()
+                => CompilationOptions ?? base.CreateCompilationOptions();
+        }
+
         private static readonly TestParameters CSharp6 =
```

Comment:
i've effectively bridged the old style of calling tests into the new test framework.  this allowed the majority of tests to remain untouched, and only tests that need actual changing/updating are affected in this PR.